app/core/config.py

```
@@ -182,6 +207,28 @@ def validate_required_and_storage(self) -> "Settings":
+        if self.connector_email_enabled and not self.connector_email_tenant_id:
+            raise ValueError(
+                "CONNECTOR_EMAIL_TENANT_ID is required when CONNECTOR_EMAIL_ENABLED=true."
+            )
+        if self.connector_file_watch_enabled and not self.connector_file_watch_path:
+            raise ValueError(
+                "CONNECTOR_FILE_WATCH_PATH is required when CONNECTOR_FILE_WATCH_ENABLED=true."
+            )
+        if self.connector_cdc_postgres_enabled and not self.connector_cdc_postgres_dsn:
+            raise ValueError(
+                "CONNECTOR_CDC_POSTGRES_DSN is required when CONNECTOR_CDC_POSTGRES_ENABLED=true."
+            )
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)